### PR TITLE
[LV] Use llvm::all_of in LoopVectorizationCostModel::getMaximizedVFForTarget. NFC

### PR DESCRIPTION
**Pull request by topperc**

### CODE REVIEW

**Comment by npanchen** [comment]
add NFC to title?

**Comment by topperc** [comment]
> add NFC to title?

Isn't it already at the end of the title?

File: llvm/lib/Transforms/Vectorize/LoopVectorize.cpp [hunk 1]
```diff
@@ -4589,15 +4589,12 @@ ElementCount LoopVectorizationCostModel::getMaximizedVFForTarget(
...
-      for (auto &pair : RUs[i].MaxLocalUsers) {
-        unsigned TargetNumRegisters = TTI.getNumberOfRegisters(pair.first);
-        if (pair.second > TargetNumRegisters)
-          Selected = false;
-      }
-      if (Selected) {
-        MaxVF = VFs[i];
+    for (int I = RUs.size() - 1; I >= 0; --I) {
+      const auto &MLU = RUs[I].MaxLocalUsers;
+      if (llvm::all_of(MLU, [&](decltype(MLU.front()) &LU) {
```
Comment (fhahn) [inline llvm/lib/Transforms/Vectorize/LoopVectorize.cpp:4594]: nit: `llvm::` should not be needed

**Review by fhahn — APPROVED** [review]
LGTM, thanks!

**Comment by npanchen** [comment]
> > add NFC to title?
> 
> Isn't it already at the end of the title?

my bad. Was looking at the beginning of the title

**Review by npanchen — APPROVED** [review]

**Pull request merged**